oe_theme.theme

```diff
@@ -977,6 +977,52 @@ function oe_theme_preprocess_paragraph__oe_text_feature_media(array &$variables)
   $cacheability->applyTo($variables);
 }
 
+/**
+ * Implements hook_preprocess_HOOK().
+ */
+function oe_theme_preprocess_paragraph__oe_iframe_media(&$variables) {
+  /** @var \Drupal\paragraphs\Entity\Paragraph $paragraph */
+  $paragraph = $variables['paragraph'];
+
+  // Get the full width value.
+  $variables['full_width'] = (bool) $paragraph->get('field_oe_iframe_media_full_width')->value;
+
+  /** @var \Drupal\media\Entity\Media $media */
+  $media = $paragraph->get('field_oe_media')->entity;
+  if (!$media instanceof MediaInterface) {
+    // The media entity is not available anymore, bail out.
+    return;
+  }
+
+  // Caches are handled by the formatter usually. Since we are not rendering
+  // the original render arrays, we need to propagate our caches to the
+  // paragraph template.
+  $cacheability = CacheableMetadata::createFromRenderArray($variables);
+  $cacheability->addCacheableDependency($media);
+  $media_type = \Drupal::entityTypeManager()->getStorage('media_type')->load($media->bundle());
+  $cacheability->addCacheableDependency($media_type);
```

Comment:
I don't think so. I add cache metadata of the media image even if user doesn't have access to that image because we don't know exactly why user doesn't have access. Might be he doesn't have access to the all media images.